client/asset/btc/btc.go

```diff
@@ -1166,132 +1183,336 @@ func (btc *ExchangeWallet) LocktimeExpired(contract dex.Bytes) (bool, time.Time,
 	return bestBlockMedianTime.After(contractExpiry), contractExpiry, nil
 }
 
-// FindRedemption attempts to find the input that spends the specified output,
-// and returns the secret key if it does. The inputs are the txid and vout of
-// the contract output that was redeemed. This method only works with contracts
-// sent from the wallet.
-func (btc *ExchangeWallet) FindRedemption(ctx context.Context, coinID dex.Bytes) (dex.Bytes, error) {
+// FindRedemption starts a goroutine to attempt finding the inputs that spends
+// the specified coin, and return the secret key for the contract when it does.
+// The goroutine scans every input of every block starting at the block in which
+// the contract was mined up till the current best block.
+// Redemption search is also triggered when a tip change is detected, to handle
+// cases where the contract is redeemed in a transaction mined after the current
+// best block.
+// The returned channel is used to notify the caller when the secret key is found
+// or if an error occurs during the search.
+//
+// NOTE: FindRedemption is necessary to deal with the case of a maker
+// redeeming but not forwarding their redemption information. The DEX does not
+// monitor for this case. While it will result in the counter-party being
+// penalized, the input still needs to be found so the swap can be completed.
+// This could potentially be an expensive operation if performed long after
+// the swap is broadcast. Realistically, though, the taker should start
+// looking for the maker's redemption beginning at swapconf confirmations
+// regardless of whether the server sends the 'redemption' message or not.
+func (btc *ExchangeWallet) FindRedemption(coinID dex.Bytes) (chan *asset.FindRedemptionResult, error) {
+	btc.findRedemptionMtx.Lock()
+	defer btc.findRedemptionMtx.Unlock()
+
 	txHash, vout, err := decodeCoinID(coinID)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("cannot decode contract coin id: %v", err)
 	}
-	txid := txHash.String()
-	// Get the wallet transaction.
-	tx, err := btc.wallet.GetTransaction(txHash.String())
+	contractOp := newOutPoint(txHash, vout)
```

Comment:
I'd like `contract` in the variable name for better readability. It can become a head scratcher reading the code later, finding references to `pt` and wondering what `pt` is.

I'd actually even favour `contractOutpoint` rather than `contractOp` as the first thing that generally comes to mind when I see `op` in programming is `operation`.

Can we make an exception for this?

Reply:
It's just that `op` is used throughout for the `output` type.

Reply:
Oh. Didn't realize `op` was already associated with the `output` type. I've renamed all `contractOp`s to `contractOutpoint`s to prevent confusing `contractOp` for `contract output` while maintaining readability. I tried using just `pt` and it was vague in certain usage sites, while `contractPt` sounds like `contract point`.